internal/util-complete/src/main/scala/sbt/internal/util/LineReader.scala

```diff
@@ -119,8 +119,9 @@ private[sbt] object JLine {
    */
   def usingTerminal[T](f: jline.Terminal => T): T =
     withTerminal { t =>
+      val result = f(t)
```

Comment:
The Javadoc comment says "This ensures synchronized access as well as re-enabling echo after getting the Terminal.", so I am guessing that you'd need to call `t.restore` both before and after `f(t)`.